packages/grafana-ui/src/components/Table/TableCellInspectModal.tsx

```diff
@@ -14,9 +14,16 @@ interface TableCellInspectModalProps {
 export function TableCellInspectModal({ value, onDismiss, mode }: TableCellInspectModalProps) {
   let displayValue = value;
   if (isString(value)) {
-    try {
-      value = JSON.parse(value);
-    } catch {} // ignore errors
+    if (value[0] === '{' || value[0] === '[' || mode === 'code') {
```

Comment:
You might want to trim `value` as `" {}"` would be valid JSON but not start with those chars.